File: 1548/1548.c
int theMostSimilarPathInAGraph(struct TreeNode* root) {
    if (!root) return 0;
    int l = theMostSimilarPathInAGraph(root->left);
    int r = theMostSimilarPathInAGraph(root->right);
    return 1 + (l > r ? l : r);
}

File: 1548/1548.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var theMostSimilarPathInAGraph = function(root) {
    if (!root) return 0;
    return 1 + Math.max(theMostSimilarPathInAGraph(root.left), theMostSimilarPathInAGraph(root.right));
};